src/App.tsx

```diff
@@ -1,61 +1,38 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import { TodosWithUsers } from './type.ts/TodosWithUser';
+import { TodoForm } from './components/TodoForm';
+import { getUserById } from './utils/userId';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+
+const todos: TodosWithUsers[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId) || null,
+}));
 
 export const App = () => {
-  return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <form action="/api/todos" method="POST">
-        <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
-        </div>
-
-        <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>
-              Choose a user
-            </option>
-          </select>
-
-          <span className="error">Please choose a user</span>
-        </div>
+  const [todosArr, setTodosArr] = useState(todos);
 
-        <button type="submit" data-cy="submitButton">
-          Add
-        </button>
-      </form>
+  const addTodo = (newTodo: TodosWithUsers) => {
+    const maxId =
+      todosArr.length > 0 ? Math.max(...todosArr.map(todo => todo.id)) : 0;
+    const currentTodo = {
+      ...newTodo,
+      id: maxId + 1,
+    };
 
-      <section className="TodoList">
-        <article data-id="1" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
+    setTodosArr(currentTodos => [...currentTodos, currentTodo]);
+  };
 
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="15" className="TodoInfo TodoInfo--completed">
-          <h2 className="TodoInfo__title">delectus aut autem</h2>
-
-          <a className="UserInfo" href="mailto:Sincere@april.biz">
-            Leanne Graham
-          </a>
-        </article>
-
-        <article data-id="2" className="TodoInfo">
-          <h2 className="TodoInfo__title">
-            quis ut nam facilis et officia qui
-          </h2>
+  return (
+    <div className="App">
+      <h1>Add todo form</h1>
 
-          <a className="UserInfo" href="mailto:Julianne.OConner@kory.org">
-            Patricia Lebsack
-          </a>
-        </article>
-      </section>
+      <TodoForm onAdd={addTodo} users={usersFromServer}/>
```

Comment:
Importing `usersFromServer` directly in the `TodoForm.tsx` component violates the guideline of preparing data in one place. Please refactor to pass the users as a prop from the `App` component.